File: src/main/java/org/snake/game/Board.java
package org.snake.game;

import org.snake.models.GameMatrix;
import org.snake.views.FigureView;

import javax.swing.*;
import javax.swing.border.EtchedBorder;
import java.awt.*;

public class Board extends JPanel {
	public static final long serialVersionUID = 1L;
	public static final Color BOARD_DEFAULT_COLOR = new Color(0, 128, 0);
	public static final int GRID_THICKNESS = 25;

	private final GameMatrix matrix;

	public Board(GameMatrix matrix) {
		super();
		this.matrix = matrix;
		this.setBackground(BOARD_DEFAULT_COLOR);
		setBorder(new EtchedBorder());
		setLayout(null);
		setSize(GRID_THICKNESS * matrix.width(), GRID_THICKNESS * matrix.height());
		setPreferredSize(new Dimension(GRID_THICKNESS * matrix.width(), GRID_THICKNESS * matrix.height()));
	}

	@Override
	protected void paintComponent(Graphics canvas) {
		super.paintComponent(canvas);
		for (int y = 0; y < matrix.height(); y++) {
			for (int x = 0; x < matrix.width(); x++) {
				matrix.get(x, y).draw(canvas, x * GRID_THICKNESS, y * GRID_THICKNESS, GRID_THICKNESS, GRID_THICKNESS);
			}
		}
	}

	public void clearLabyrinth(){
		for(int y = 0; y < matrix.height(); y++){
			for(int x = 0; x < matrix.width(); x++){
				matrix.removeView(x, y);
			}	
		}
	}
}

File: src/main/kotlin/org/snake/game/Board.kt
package org.snake.game

import org.snake.models.GameMatrix
import java.awt.Color
import java.awt.Dimension
import java.awt.Graphics
import javax.swing.JPanel
import javax.swing.border.EtchedBorder

class Board(private val matrix: GameMatrix) : JPanel() {
    override fun paintComponent(canvas: Graphics) {
        super.paintComponent(canvas)
        for (y in 0 until matrix.height()) {
            for (x in 0 until matrix.width()) {
                matrix[x, y]!!.draw(canvas, x * GRID_THICKNESS, y * GRID_THICKNESS, GRID_THICKNESS, GRID_THICKNESS)
            }
        }
    }

    fun clearLabyrinth() {
        for (y in 0 until matrix.height()) {
            for (x in 0 until matrix.width()) {
                matrix.removeView(x, y)
            }
        }
    }

    companion object {
        const val serialVersionUID = 1L
        val BOARD_DEFAULT_COLOR = Color(0, 128, 0)
        const val GRID_THICKNESS = 10
    }

    init {
        background = BOARD_DEFAULT_COLOR
        border = EtchedBorder()
        layout = null
        setSize(GRID_THICKNESS * matrix.width(), GRID_THICKNESS * matrix.height())
        preferredSize = Dimension(GRID_THICKNESS * matrix.width(), GRID_THICKNESS * matrix.height())
    }
}